compiler/rustc_privacy/src/lib.rs

```diff
@@ -988,10 +988,7 @@ impl<'tcx> Visitor<'tcx> for NamePrivacyVisitor<'tcx> {
     fn visit_expr(&mut self, expr: &'tcx hir::Expr<'tcx>) {
         if let hir::ExprKind::Struct(qpath, fields, ref base) = expr.kind {
             let res = self.typeck_results().qpath_res(qpath, expr.hir_id);
-            let Some(adt) = self.typeck_results().expr_ty(expr).ty_adt_def() else {
-                self.tcx.dcx().span_delayed_bug(expr.span, "no adt_def for expression");
-                return;
-            };
+            let adt = self.typeck_results().expr_ty(expr).ty_adt_def().unwrap();
```

Comment:
Slightly concerned that this will lead to an ICE that used not to happen (`span_delayed_bug` has been the go-to solution for "this is invalid state but other errors will likely have been emitted so an ICE isn't necessary).

Reply:
It's just reverting to what was there before #121482.